src/components/Workflow/SearchableWorkflowNameList.tsx

```diff
@@ -11,27 +11,93 @@ import { WorkflowExecutionPhase } from 'models/Execution/enums';
 import { Shimmer } from 'components/common/Shimmer';
 import { WorkflowExecutionIdentifier } from 'models/Execution/types';
 import { debounce } from 'lodash';
-import { Typography } from '@material-ui/core';
+import {
+  IconButton,
+  Typography,
+  FormControlLabel,
+  Checkbox,
+  FormGroup,
+  Button,
+  CircularProgress,
+} from '@material-ui/core';
+import UnarchiveOutline from '@material-ui/icons/UnarchiveOutlined';
+import ArchiveOutlined from '@material-ui/icons/ArchiveOutlined';
+import { useMutation } from 'react-query';
+import { WorkflowExecutionState } from 'models/Workflow/enums';
+import { updateWorkflowState } from 'models/Workflow/api';
+import { useState } from 'react';
+import { useSnackbar } from 'notistack';
 import { WorkflowListStructureItem } from './types';
 import ProjectStatusBar from '../Project/ProjectStatusBar';
 import { workflowNoInputsString } from '../Launch/LaunchForm/constants';
 import { SearchableInput } from '../common/SearchableList';
 import { useSearchableListState } from '../common/useSearchableListState';
 import { useWorkflowInfoItem } from './useWorkflowInfoItem';
+import t from '../Executions/Tables/WorkflowExecutionTable/strings';
+import { getArchiveStateString, isWorkflowArchived } from './utils';
 
 interface SearchableWorkflowNameItemProps {
   item: WorkflowListStructureItem;
 }
 
+interface SearchableWorkflowNameItemActionsProps {
+  item: WorkflowListStructureItem;
+  setHideItem: (hide: boolean) => void;
+}
+
 interface SearchableWorkflowNameListProps {
   workflows: WorkflowListStructureItem[];
+  onArchiveFilterChange: (showArchievedItems: boolean) => void;
+  showArchived: boolean;
 }
 
+export const showOnHoverClass = 'showOnHover';
+
 const useStyles = makeStyles(() => ({
+  actionContainer: {
+    display: 'block',
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    height: '100%',
+  },
+  actionProgress: {
+    width: '100px',
+    textAlign: 'center',
+    top: '42%',
+    display: 'block',
+    position: 'absolute',
+    right: 0,
+  },
+  archiveButton: {
+    right: '30px',
+    position: 'relative',
+    top: '42%',
+    height: 'auto',
+  },
+  archiveCheckbox: {
+    whiteSpace: 'nowrap',
+  },
+  confirmationBox: {
+    height: '100%',
+    [`& > button`]: {
+      height: '100%',
+    },
+  },
+  confirmationButton: {
+    borderRadius: 0,
+    minWidth: '100px',
+    minHeight: '53px',
```

Comment:
I'm not a big fan of using 100% in calculating size of the items, as it could easily backfire + any change in the size of the item, would need to be accompanied with child changes (for cases like top: 42%). I feel like it is better to use flex in modern browsers. As we do not plan out users to use IE6 and mainly use chromium as a guideline.
My suggestion would be, update styles to something like:
```
const useStyles = makeStyles((theme: Theme) => ({  // <- Add theme parameters
  actionContainer: {
    display: 'flex',
    right: 0,
    top: 0,
    position: 'absolute',
    height: '100%',
overflow: hidden;
    border-radius: 0px 16px 16px 0px; // to ensure that cancel button will have rounded corners on the right side
  },
 centeredChild: { // <- new one applied along side actionContainer for Loading state or
                             //Archive/Unarchive icon button
    alignItems: 'center';
    padding: theme.spacing(2);
},
  confirmationButton: {
minWidth: '100px',
border-radius: 0
}

```
remove `actionProgress` and `archiveButton` styles, as they would be covered by applying `centeredChild` style to
remove ` <div className={styles.confirmationBox}>` and replace it with `<> </>`
replace <div className={styles.actionProgress}> with <div className={classNames(styles.actionContainer, centeredChild)}> (not sure if we should add onHover here too - your call.